c++/src/kj/async-io.h

```diff
@@ -169,6 +169,22 @@ class AsyncIoStream: public AsyncInputStream, public AsyncOutputStream {
   virtual kj::Maybe<int> getFd() const { return nullptr; }
   // Get the underlying Unix file descriptor, if any. Returns nullptr if this object actually
   // isn't wrapping a file descriptor.
+
+  virtual Maybe<void*> getWin32Handle() const { return nullptr; }
+  // Get the underlying Win32 HANDLE, if any. Returns nullptr if this object actually isn't
+  // wrapping a handle.
+
+  static Maybe<int> releaseFd(Own<AsyncIoStream>&& stream);
```

Comment:
It's a little unexpected that this is a static method with a private virtual method for implementation. I get that the intent is that it forces you to destroy the stream. But what if the function returns null? Then you've inadvertently lost the stream altogether.

I think possibly we need to make this an instance method with documentation saying "if this returns non-null, you must immediately destroy the stream object before using the FD".